torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6836,6 +6852,22 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
         supports_out=False,
         sample_inputs_func=sample_inputs_logdet,
         decorators=(skipCPUIfNoLapack, skipCUDAIfNoMagma, skipCUDAIfRocm)),
+    # `log_softmax` supports different dtypes based on whether `dtype` argument,
+    # is passed or not. Hence two OpInfo entries, one with dtype and other without.
+    OpInfo(
+        'log_softmax',
+        supports_out=False,
+        dtypes=floating_types_and(torch.bfloat16),
+        dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
+        sample_inputs_func=sample_inputs_log_softmax,
+        assert_autodiffed=True),
+    OpInfo(
+        'log_softmax',
+        variant_test_name='dtype',
+        supports_out=False,
+        dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
```

Comment:
Not sure, as to how complex is also supported with `dtype=torch.float64`.

Reply:
Looks like this part of code facilitates that,
https://github.com/pytorch/pytorch/blob/c3bf42e0d8b0a820a0fc762e520de4a6377562a3/aten/src/ATen/native/SoftMax.cpp#L273-L274

```python
>>> t = torch.randn(3,3, dtype=torch.complex128)
>>> torch.log_softmax(t, 0, dtype=torch.float64)
<stdin>:1: UserWarning: Casting complex values to real discards the imaginary part (Triggered internally at  ../aten/src/ATen/native/Copy.cpp:240.)
```

It does raise a warning. But I think it should just error out.

Reply:
This is really interesting -- thanks for taking a closer look. I don't think we should take any action for this behavior in this PR, however.

fyi @anjali411